sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -283,7 +284,7 @@ class HadoopTableReader(
       _minSplitsPerRDD)
 
     // Only take the value (skip the key) because Hive works only with values.
-    rdd.map(_._2)
+    CombineSmallFile.combineWithPath(rdd, sc, path).map(_._2)
```

Comment:
@liancheng We can do this except `S3`, or do you have good idea about listing files in `S3`?